### ISSUE TITLE
Issues with Intel GPU Drivers

### ISSUE THREAD

**Issue by AniLeo** [RPCS3/rpcs3#6199]
Intel proprietary drivers are currently not supported by RPCS3 as they're too broken, specially on OpenGL

**This is a megathread for issues with Intel Drivers**, as there's no need to have several wontfix issues on this where RPCS3 isn't even at fault

---

- [GLSL: 'r8' assumed as being a reserved word while it isn't](https://github.com/RPCS3/rpcs3/issues/3313)
- [`h0.w = clamp((tc2 * fc48), 0., 1.).w;` saturate fails to compile](https://github.com/RPCS3/rpcs3/issues/4342)
- [Driver returning tex00 instead of tex0](https://github.com/RPCS3/rpcs3/issues/5690)

**Comment by d-apps** [comment]
If anyone have "**RSX: Failed to compile shader**" errors if you use the specific build [**0.0.5-7550**](https://github.com/RPCS3/rpcs3-binaries-win/releases/download/build-32059bfaa2aeab9dece0a32d196cddb6622e2994/rpcs3-v0.0.5-7550-32059bfa_win64.7z) this error might not happen, at least for me, looks like only after this build version the changes that was made break the "RSX Decompiler Thread" making it **fails to compile shaders** for Intel iGPUs.

You won't have the new features from the newest builds :/ but at least you can play some games wthout that error.

![1](https://user-images.githubusercontent.com/20110433/66060932-d8263500-e514-11e9-8228-59ed0c3d5c05.PNG)

**Comment by Matterhorn56** [comment]
I'm looking to use mesa3d
("Open source implementations of OpenGL, OpenGL ES, Vulkan, OpenCL, and more!", https://www.mesa3d.org)
but I would like to know what is the minimum required version of vulkan.
Thanks

**Comment by AniLeo** [comment]
Specific features and extensions are required, can't measure by "version"
Use latest mesa and it will work

**Comment by Matterhorn56** [comment]
> Use latest mesa and it will work

That's great to hear.

**Comment by AniLeo** [comment]
It would be interesting to know if something changed regarding the GLSL compiler on newer Intel Windows drivers.

**Comment by kd-11** [comment]
Many years ago I implemented GL_ARB_Spirv for OpenGL then lost the code. A very dumb thing to happen, I'll have to find them and try to rebase on top of current renderer arch but I remember results were promising. My mistake back then was to try and use SPIRV for all shaders when in fact it is only needed for the complex shaders that are emitted by the recompiler. We do not need them for things like texture decoding which use basic glsl anyway.
